core/options.js

```diff
@@ -31,57 +31,57 @@ goog.requireType('Blockly.WorkspaceSvg');
  * @constructor
  */
 Blockly.Options = function(options) {
-  var readOnly = !!options['readOnly'];
-  if (readOnly) {
-    var toolboxJsonDef = null;
-    var hasCategories = false;
-    var hasTrashcan = false;
-    var hasCollapse = false;
-    var hasComments = false;
-    var hasDisable = false;
-    var hasSounds = false;
-  } else {
-    var toolboxJsonDef = Blockly.utils.toolbox.convertToolboxDefToJson(options['toolbox']);
-    var hasCategories = Blockly.utils.toolbox.hasCategories(toolboxJsonDef);
-    var hasTrashcan = options['trashcan'];
+  let toolboxJsonDef = null;
+  let hasCategories = false;
+  let hasTrashcan = false;
+  let hasCollapse = false;
+  let hasComments = false;
+  let hasDisable = false;
+  let hasSounds = false;
+  const readOnly = !!options['readOnly'];
+  if (!readOnly) {
+    toolboxJsonDef = Blockly.utils.toolbox.convertToolboxDefToJson(options['toolbox']);
+    hasCategories = Blockly.utils.toolbox.hasCategories(toolboxJsonDef);
+    hasTrashcan = options['trashcan'];
     if (hasTrashcan === undefined) {
       hasTrashcan = hasCategories;
     }
-    var maxTrashcanContents = options['maxTrashcanContents'];
-    if (hasTrashcan) {
-      if (maxTrashcanContents === undefined) {
-        maxTrashcanContents = 32;
-      }
-    } else {
-      maxTrashcanContents = 0;
-    }
-    var hasCollapse = options['collapse'];
+    hasCollapse = options['collapse'];
     if (hasCollapse === undefined) {
       hasCollapse = hasCategories;
     }
-    var hasComments = options['comments'];
+    hasComments = options['comments'];
     if (hasComments === undefined) {
       hasComments = hasCategories;
     }
-    var hasDisable = options['disable'];
+    hasDisable = options['disable'];
     if (hasDisable === undefined) {
       hasDisable = hasCategories;
     }
-    var hasSounds = options['sounds'];
+    hasSounds = options['sounds'];
     if (hasSounds === undefined) {
       hasSounds = true;
     }
   }
-  var rtl = !!options['rtl'];
-  var horizontalLayout = options['horizontalLayout'];
+
+  let maxTrashcanContents = options['maxTrashcanContents'];
+  if (hasTrashcan) {
+    if (maxTrashcanContents === undefined) {
+      maxTrashcanContents = 32;
+    }
+  } else {
+    maxTrashcanContents = 0;
+  }
+  const rtl = !!options['rtl'];
+  let horizontalLayout = options['horizontalLayout'];
```

Comment:
Did you move `maxTrashcanContents` in order to alphabetise the setting of the various local variables?  If so then shouldn't it go _after_ `horizontalLayout`?

Reply:
I moved it out of the if statement so that `maxTrashcanContents` would always be defined. 

Before, if `readOnly = true`, then `maxTrashcanContents = undefined`.
Now, if `readOnly = true`, then `maxTrashcanContents = 0` (because `hasTrashcan = false`.

I don't expect this to change any behavior, but it fixes typing issues with `maxTrashcanContents`.

Reply:
Oh yes, quite so.  Apologies for my misreading.